### DISCUSSION TITLE
interrupts enabled register correct?

### DISCUSSION THREAD

**Discussion by rizacelik** [jrowberg/i2cdevlib#666] [Q&A] [answered]
Example code https://github.com/jrowberg/i2cdevlib/blob/master/Arduino/MPU6050/MPU6050_6Axis_MotionApps20.cpp#L315

#define MPU6050_INTERRUPT_FIFO_OFLOW_BIT    4
#define MPU6050_INTERRUPT_DMP_INT_BIT       1

DEBUG_PRINTLN(F("Setting DMP and FIFO_OFLOW interrupts enabled..."));
setIntEnabled(1<<MPU6050_INTERRUPT_FIFO_OFLOW_BIT|1<<MPU6050_INTERRUPT_DMP_INT_BIT);
equal
**setIntEnabled(0x12);**

0x12 => **0001 0010**
interrupts enabled register correct?

https://www.waveshare.com/w/upload/0/03/MPU6050-Register-Map-and-Descriptions.pdf

![image](https://user-images.githubusercontent.com/19993109/152496360-a4684bee-5e45-4ace-9d2c-2c4d9cca4178.png)

**Comment by jrowberg** [comment, marked as answer]
Yes, it's correct; the DMP interrupt is not documented, but it's there. If you disable bit 1, you'll find that the DMP packet interrupt is no longer working.

**Reply by rizacelik** [reply]
Thank you for your answer, if there is a register map file, can you post a link?

**Reply by ZHomeSlice** [reply]
this is more difficult to find
Here is the code provided by InvenSense to do the DMP Interrupt enable for V6.12
V6.12 development package:
motion_driver_6.12.zip
inv_mpu.c  on line 527
[https://invensense.tdk.com/developers/](https://invensense.tdk.com/developers/)
```
#define BIT_DMP_INT_EN      (0x02)
//and 
// elseware we have  st.reg->int_enable     = 0x38,

// the Funciton that uses it:
/**
 *  @brief      Enable/disable data ready interrupt.
 *  If the DMP is on, the DMP interrupt is enabled. Otherwise, the data ready
 *  interrupt is used.
 *  @param[in]  enable      1 to enable interrupt.
 *  @return     0 if successful.
 */
static int set_int_enable(unsigned char enable)
{
    unsigned char tmp;
    if (st.chip_cfg.dmp_on) {
        if (enable)
            tmp = BIT_DMP_INT_EN; // <<<<<< we want interrupt enabled for DMP
        else
            tmp = 0x00;
        if (i2c_write(st.hw->addr, st.reg->int_enable, 1, &tmp)) // <<<< Set it here
            return -1;
        st.chip_cfg.int_enable = tmp;
    } else {
        if (!st.chip_cfg.sensors)
            return -1;
        if (enable && st.chip_cfg.int_enable)
            return 0;
        if (enable)
            tmp = BIT_DATA_RDY_EN;
        else
            tmp = 0x00;
        if (i2c_write(st.hw->addr, st.reg->int_enable, 1, &tmp))
            return -1;
        st.chip_cfg.int_enable = tmp;
    }
    return 0;
}
```
Z

**Reply by rizacelik** [reply]
Thank you for your valuable contributions.
